web/blog/native-zeek-log-rotation-and-shipping/index.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Native Zeek log Rotation & Shipping
+authors: mavam
+date: 2023-07-27
+tags: [zeek, logs, shipping, rotation, pipelines]
+comments: true
+---
+
+Did you know that [Zeek](http://zeek.org) support log rotation triggers, so that
+you can do anything you want with a newly rotated batch of logs?
+
+![Zeek Log Rotation](zeek-log-rotation.excalidraw.svg)
+
+<!-- truncate -->
+
+This blog post shows you how to use Zeek's native log rotation feature to
+conveniently invoke any post-processor, such as a log shipper. We illustrate how
+you can ingest data into Tenzir, but you can plug in any log shipper you want.
+
+## External Log Shipping (pull)
+
+In case you're not using Zeek's native log rotation trigger, you may observe a
+directory to which Zeek periodically writes files.
+
+Log shippers can take care of that, but your mileage may vary. For example,
+[Filebeat][filebeat] works for stock Zeek only. Every log file is hard-coded. If
+you have custom scripts or extend some logs, you're left alone. Filebeat also
+uses the stock Zeek JSON output, which has no type information. Filebeat then
+brings the typing back manually later as it converts the logs to the Elastic
+Common Schema (ECS).
```

Comment:
The shortcoming of filebeat here is that the Zeek integration lacks  modularity. 

But for this tool in particular, you would not monitor the rotated logs but the live ones instead. Filebeat tracks a read offset for each log file internally and implements precautions to prevent double-shipping of log lines when the files get rotated. One of the advantages is a much lower latency compared to rotation based shipment. The lack of support for user or script defined logs is really what is hampering filebeat in particular, but a similar tool that does not insist on a particular taxonomy would certainly be a viable alternative.

I think what I'm trying to say is that filebeat is not a great example because of the problems with ECS.